variants/bpi-bit/pins_arduino.h

```diff
@@ -0,0 +1,63 @@
+#ifndef Pins_Arduino_h
+#define Pins_Arduino_h
+
+#include <stdint.h>
+
+#define EXTERNAL_NUM_INTERRUPTS 16
+#define NUM_DIGITAL_PINS 40
+#define NUM_ANALOG_INPUTS 16
+
+#define analogInputToDigitalPin(p) (((p) < 20) ? (esp32_adc2gpio[(p)]) : -1)
```

Comment:
This define is used to convert from A[n] to gpio pin.  Since you are choosing not to define analog input pins should you change both `NUM_ANALOG_INPUTS` and `analogInputtoDigitalPin()` to reflect this choice?
```c++
#define NUM_ANALOG_INPUTS 0
#define analogInputToDigitalPin(p) -1
```

Chuck.

Reply:
I think the definition here can be retained, because we not only have to consider the primary users, but also the people with certain Arduino and ESP32 programming development foundations, they can not use our prepared P in their own code ( n), from the line definition A (n) such a pin, and, our board does not give up the ADC function.

yelvlab

Reply:
I think the definition here can be retained, because we not only have to consider the primary users, but also the people with certain Arduino and ESP32 programming development foundations, they can not use our prepared P(n) in their own code , from the line definition A (n) such a pin, and, our board does not give up the ADC function.